extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -192,7 +192,7 @@ function ScrollerWithRef(
   // Trigger render by setting the resting index to the current scroll state.
   const debouncedResetScrollReferencePoint = useMemo(() => {
     // Use local window if possible.
-    const win = containerRef.current ? getWin(containerRef.current) : window;
+    const win = getWin(containerRef.current);
     return debounce(
       win,
```

Comment:
While debounce itself is pure, the function that it returns is bound to the window instance when it was created. When that function is invoked, it'll try to dereference a `win.Date`, and if `win` is null, it'll throw.

What I think needs to happen is `debouncedResetScrollReferencePoint` should itself be put inside a `useLayoutEffect`. We can use a `useRef` to ensure stability of the function instance.

Reply:
I do believe that `useMemo` is the correct hook to use here. I feel like creating a ref and using a hook to update the ref is redundant.

What I think we can do is create a condition if win is undefined we can just execute an empty function that doesn't do anything.

Reply:
As long as `win` is closed over, and `useMemo` doesn't take into account that `containerRef.current` has changed from `null` to any node, then we'll have a broken debounce that can't actually execute. You can hoist this out and add `window` as part of the memo:

```js
const win = getWin(containerRef.current);
const debouncedResetScrollReferencePoint = useMemo(() => {
  return debounce(win, …);
}, [restingIndex, setRestingIndex, win]);
```

(Also, debounce fires when the timer ends, but we may have cleaned up the memo here. That's a bug crying for a uniform `useDebounceCallback` helper)